### ISSUE TITLE
Docker daemon logs timestamp format

### ISSUE THREAD

**Issue by Crassirostris** [docker/docker#28608]
**Description**

Recently in Kubernetes we encountered a problem with docker daemon logs https://github.com/kubernetes/kubernetes/issues/36893. Turns out, some log lines have different timestamp format from one we relied on. Example:

```
time="2016-11-15T14:24:48.169947487Z" level=debug msg="Sending 15 to ab086567fdc3a16399936f40647ea4770ecc699dd201da60f1e683a882a1cc58" 
time="2016-11-15T14:24:48.184869903Z" level=debug msg="Calling GET /v1.23/images/json" 
time="2016-11-15T14:24:48Z" level=debug msg="containerd: process exited" id=ab086567fdc3a16399936f40647ea4770ecc699dd201da60f1e683a882a1cc58 pid=init status=0 systemPid=31321 
time="2016-11-15T14:24:48.235490652Z" level=debug msg="received containerd event: &types.Event{Type:\"exit\", Id:\"ab086567fdc3a16399936f40647ea4770ecc699dd201da60f1e683a882a1cc58\", Status:0x0, Pid:\"init\", Timestamp:0x582b1ab0}" 
time="2016-11-15T14:24:48.235666690Z" level=warning msg="container ab086567fdc3a16399936f40647ea4770ecc699dd201da60f1e683a882a1cc58 restart canceled" 
```

As you can see, third message have only seconds precision. The full log can be found [there](https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-soak-continuous-e2e-gce/6776/artifacts/jenkins-e2e-minion-group-t4v6/docker.log), it's a e2e k8s test.

The question is, does it work as intended or is it a bug?

**Output of `docker version`:**

```
$ docker --version
Docker version 1.11.2, build 4dc5990
```

**Comment by LK4D4** [comment]
I think @mlaventure changed the format of containerd logs some time ago. However, it's still different from docker, but should look almost the same.

**Comment by mlaventure** [comment]
It was changed back in April to use `RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"`:

```diff
diff --git a/containerd/main.go b/containerd/main.go
index 8a78328..1d621c7 100644
--- a/containerd/main.go
+++ b/containerd/main.go
@@ -72,6 +72,7 @@ var daemonFlags = []cli.Flag{
 }
 
 func main() {
+       logrus.SetFormatter(&logrus.TextFormatter{TimestampFormat: time.RFC3339Nano})
        appendPlatformFlags()
        app := cli.NewApp()
        app.Name = "containerd"
```

Docker seems to be using an altered version `rFC3339NanoLocal = "2006-01-02T15:04:05.999999999"`

**Comment by thaJeztah** [comment]
@mlaventure should this issue be moved to the containerd issue tracker?

**Comment by mlaventure** [comment]
@thaJeztah I don't think so. The fix is already present in 1.12.x

**Comment by Crassirostris** [comment]
@mlaventure 

> I don't think so. The fix is already present in 1.12.x

OK, great, thanks! Closing the issue then

**Comment by thaJeztah** [comment]
Thanks for reporting @Crassirostris !
